README: updated to new architecture (#456)
redhatcloudx/cloud-image-directory: README.md

# Cloud Image Directory

> This is an experiment project is a work in progress with a goal of making it easier

Reviewer: ```suggestion
This is an experimental project and a work in progress with the goal of making it easier
```

> to find Red Hat Enterprise Linux™, Fedora and in the future a bunch of other linux images

Reviewer: ```suggestion
to find Red Hat Enterprise Linux™, Fedora and, in the future, a bunch of other linux images
```

at various public cloud providers.

## Components

This repository contains the tool `cloudimagedirectory-transformer`, which is used to transform the unstructured data provided by `cloud-image-retriever` into a predefined schema for the `directory-frontend`.

> It belongs to three other sub projects.

Author: ```suggestion
It belongs to three other sub projects:
````

- [cloud-image-retriever](https://github.com/redhatcloudx/cloud-image-retriever): Used to retrieve unstructured image data from public cloud providers.

> - [infrastructure](https://github.com/redhatcloudx/infrastructure): Defines the public cloud infrastructure [imagedirectory.cloud](https://imagedirectory.cloud/) needs.

Reviewer: ```suggestion
- [infrastructure](https://github.com/redhatcloudx/infrastructure): Defines the public cloud infrastructure [imagedirectory.cloud](https://imagedirectory.cloud/).
```

- [directory-frontend](https://github.com/redhatcloudx/cloud-image-directory-frontend) Used to visualize the dataset.

### Architecture diagram

![Architecture diagram](assets/arch-diagram.png)

> The Collector(`cloud-image-retriever`) fetches the image from different cloud providers on a predefined interval. This data gets then stored in a s3 bucket.

Reviewer: ```suggestion
The Collector(`cloud-image-retriever`) fetches the images from different cloud providers on a predefined interval. This data is then stored in a s3 bucket.
```

> After a successful collection, the transformer(`cloudimagedirectory-transformer`) reads the unstructured image data from the S3 bucket and generates static image content in a consumable format. The transformer generates also lookup tables to allow an efficient search from e.g. a js single page application. Once the transformation is complete, it's written back into the s3 bucket.

Reviewer: ```suggestion
After a successful collection, the transformer(`cloudimagedirectory-transformer`) reads the unstructured image data from the S3 bucket and generates static image content in a consumable format. The transformer also generates lookup tables to allow for efficient searches from e.g. a js single page application. Once the transformation is complete, it's written back into the s3 bucket.
```

> In parallel frontend code is deployed by putting it into the s3 bucket too. This can be done by a [release pipeline](https://github.com/redhatcloudx/cloud-image-directory-frontend/blob/main/.github/workflows/release.yaml).

Reviewer: ```suggestion
In parallel, frontend code is deployed by putting it into the s3 bucket too. This is done by a [release pipeline](https://github.com/redhatcloudx/cloud-image-directory-frontend/blob/main/.github/workflows/release.yaml).
```
